nixos/modules/services/networking/syncplay.nix

```diff
@@ -70,30 +238,30 @@ in
         '';
       };
 
+      package = mkOption {
+        type = types.package;
+        default = pkgs.syncplay-nogui;
+        defaultText = literalExpression "pkgs.syncplay-nogui";
+        description = ''
+          Package to use for syncplay.
+        '';
+      };
+
       user = mkOption {
         type = types.str;
-        default = "nobody";
+        default = "syncplay";
         description = ''
           User to use when running Syncplay.
         '';
       };
 
       group = mkOption {
         type = types.str;
-        default = "nogroup";
+        default = "syncplay";
         description = ''
           Group to use when running Syncplay.
         '';
       };
```

Comment:
Note that with `DynamicUser`, these only set the name of the user and group, and it would default to `syncplay` anyway (derived from the systemd unit name). Therefore, these options won’t work the same way they did as before, and they’re not very useful either. Two options here:

1. Use `lib.mkRemovedOptionModule` to deprecate the options.
2. Default these to `null`, and disable `DynamicUser` if they’re set, if there’s a good reason people might want to run `syncplay` as a specific persistent user (long‐term state?).

Reply:
The `--rooms-db-file` and `--stats-db-file` flags do introduce long-term state to the service which wasn't present when the original module was made. I was under the assumption that `DynamicUser` and `StateDirectory` would allow such state to persist, is that not the case?

Reply:
Yes, it will. Usually you’d only want to make a persistent user if there’s a very large tree of persistent state (since systemd may have to `chown` it at runtime), or if there’s a compelling reason to want to have a specific existing user/group own the resources (e.g., for sharing). So I suspect option 1 is the way to go here.

Reply:
Additionally, with option 2 were you thinking something like this?
```nix
      serviceConfig = rec {
        User = mkIf (cfg.user != null) cfg.user;
        Group = mkIf (cfg.group != null) cfg.group;
        DynamicUser = cfg.user == null && cfg.group == null;
        StateDirectory = mkIf DynamicUser "syncplay";
        WorkingDirectory = mkIf DynamicUser "%S/syncplay";
        [ ... ]
      };
```

Reply:
Something along those lines, yes. But `StateDirectory` and `WorkingDirectory` don’t require `DynamicUser`. Possibly you’d want to make the persistent state path options take absolute paths instead in this case and just have users specify `%S/syncplay/…` for the common case rather than setting `WorkingDirectory`; that might be a good idea in general regardless of approach taken. Anyway, there are many services that only support `DynamicUser`, and users with highly‐specialized needs can override the systemd unit settings, so I don’t think it’s worth spending much complexity supporting a theoretical use‐case.

Reply:
I think I'll go with deprecating those options then.